indico/modules/auth/controllers.py

```diff
@@ -243,7 +277,9 @@ def _process(self):
                                       User.all_emails.contains(db.func.any(list(handler.get_all_emails(form)))))
 
         if form.validate_on_submit():
-            if handler.must_verify_email:
+            if account_moderation_enabled and handler.must_verify_email:
```

Comment:
We only want account moderation for local accounts. However, `must_verify_email` may also be true for remote providers (e.g. because we don't know that they send us a verified email). Check if `isinstance(handler, LocalRegistrationHandler)` instead. (or maybe add a `is_local` attribute to the registration handlers so you can check `handler.is_local`)